src/input_xml.F90

```diff
@@ -4213,7 +4375,122 @@ subroutine read_cross_sections_xml()
+           &// trim(path_cross_sections) // "' does not exist!")
+    end if
+
+    call write_message("Reading cross sections XML file...", 5)
+
+    ! Parse cross_sections.xml file
+    call open_xmldoc(doc, path_cross_sections)
+
+    if (check_for_node(doc, "groups")) then
+      ! Get neutron group count
+      call get_node_value(doc, "groups", energy_groups)
+    else
+      call fatal_error("groups element must exist!")
+    end if
+
+    allocate(energy_bins(energy_groups + 1))
+    if (check_for_node(doc, "group_structure")) then
+      ! Get neutron group structure
+      call get_node_array(doc, "group_structure", energy_bins)
+    else
+      call fatal_error("group_structures element must exist!")
+    end if
+
+    allocate(energy_bin_avg(energy_groups))
+    do i = 1, energy_groups
+      energy_bin_avg(i) = 0.5_8 * (energy_bins(i) + energy_bins(i + 1))
+    end do
+
+    allocate(inverse_velocities(energy_groups))
+    if (check_for_node(doc, "inverse_velocities")) then
+      ! Get inverse velocities
+      call get_node_array(doc, "inverse_velocities", inverse_velocities)
+    else
+      ! If not given, estimate them by using average energy in group which is
+      ! assumed to be the midpoint
+      do i = 1, energy_groups
+        inverse_velocities(i) = &
+             (sqrt(TWO * energy_bin_avg(i) / (MASS_NEUTRON_MEV)) * &
+              C_LIGHT * 100.0_8)
+      end do
+    end if
+
+    ! Get node list of all <xsdata>
+    call get_node_list(doc, "xsdata", node_xsdata_list)
+    n_listings = get_list_size(node_xsdata_list)
+
+    ! Allocate xs_listings array
+    if (n_listings == 0) then
+      call fatal_error("No XSDATA listings present in cross_sections.xml &
+           &file!")
+    else
+      allocate(xs_listings(n_listings))
+    end if
+
+    do i = 1, n_listings
+      listing => xs_listings(i)
+
+      ! Get pointer to xsdata table XML node
+      call get_list_item(node_xsdata_list, i, node_xsdata)
+
+      ! copy a number of attributes
+      call get_node_value(node_xsdata, "name", listing % name)
+      listing % name = to_lower(listing % name)
+      listing % alias = listing % name
+      if (check_for_node(node_xsdata, "alias")) &
+           call get_node_value(node_xsdata, "alias", listing % alias)
+      listing % alias = to_lower(listing % alias)
+      if (check_for_node(node_xsdata, "zaid")) then
+        call get_node_value(node_xsdata, "zaid", listing % zaid)
+      else
+        listing % zaid = -1
+      end if
+      if (check_for_node(node_xsdata, "awr")) then
+        call get_node_value(node_xsdata, "awr", listing % awr)
+      else
+        ! Set to a default of 1; this allows a macroscopic library to still
+        ! be used with materials with atom/b-cm units for testing purposes
+        listing % awr = ONE
+      end if
+      if (check_for_node(node_xsdata, "kT")) then
+        call get_node_value(node_xsdata, "kT", listing % kT)
+      else
+        listing % kT = 2.53E-8_8
```

Comment:
If you intend to have a specific temperature, I'd rather see `293.6_8 * K_BOLTZMANN`